docs/source/Technical-Glossary.rst

```diff
@@ -0,0 +1,188 @@
+
+.. image:: https://cdnssinc-prod.softserveinc.com/img/blog/containers-security-virtual-machines.PNG
+
+image credits: `docker.com <https.docker.com>`_
+
+For a more in-depth explanation of Docker and how it works, consder looking at `this series of articles.
+<https://www.nschoe.com/articles/2016-05-26-Docker-Taming-the-Beast-Part-1.html>`_
+
+
+Postgres
+--------
+
+PostgreSQL is a general purpose and object-relational database management system. It allows you to store and query large
+amounts of data without having to worry about all of the complicated processes going on under the hood. [*]_  PostgreSQL
+optimizes data querying for you, making your application faster. All information and metadata is stored in it.
+
+
+
+
+RabbitMQ
+---------
+
+RabbitMQ is a message broker. When messages are sent online, they go from the producer to a queue and then to the
+consumer. RabbitMQ is that queue. Instead of having to perform all tasks involving sending messages, including generating
+PDFs, locating the recipient, etc., the message producer only has to upload their message and instructions to the queue
+and RabbitMQ will take care of the rest. Using this service makes messaging through Lookit easier and more efficient, as
+it is able to re-queue messages, it is faster and more reliable, and it is scalable for when there are a lot of messages.
+
+`source <https://www.cloudamqp.com/blog/2015-05-18-part1-rabbitmq-for-beginners-what-is-rabbitmq.html>`_
+
+
+Ngrok
+-----
+
+Ngrok is used in the development process to act as a tunnel into your PC. It is not secure to allow access into your PC
+or local address through a public channel, as this can open you up to malicious attacks. Ngrok allows you to securely
+provide acces to your local address through something public, like the internet. When Ngrok is running on your machine,
+it gets the port of the network service, which is usually a web server, and then connects to the Ngrok cloud service.
+This cloud service takes in traffic and requests from a public address and then relays that traffic directly to the
+Ngrok process running on your machine, which then passes it along to the local address you specify. By doing this, you
+can minimize the interaction between outside traffic and your personal machine.
+
+When trying to stream videos on the development stage, AWS will need an address to send the video to. Ngrok will create
+a dummy link for this purpose and then send the video from this dummy address to your PC.
+
+
+
+
+
+
+
+External Resources
+___________________
+
+Google Cloud
+-------------
+
+The Cloud service is where all the studies are stored.
+
+
+Amazon Web Services
+--------------------
+
+This is where the consent videos are stored??
+
+Celery
+-------
+
+This is what runs the long term tasks
+
+Authenticator
+---------------
+
+Allows you to log into your account securely
+
+Lookit Ember Frameplayer
+------------------------
+
+Consent manager videos
```

Comment:
This is actually the frameplayer that provides the functionality for the experiments themselves - parsing the JSON documents specifying the study protocol into a sequence of "frames" and running them. Can link to that code and/or install directions